### ISSUE TITLE
Default options

### ISSUE THREAD

**Issue by Kagami** [webpack/file-loader#30]
Hi.

Probably my use case is not popular or even this feature request should go to the `webpack` directly. But I want to have ability to specify default options for the file loader, e.g. set the name to `[name].[ext]`.

I have a `worker.js` code which uses Web Worker interface and want to include it to the build using the file loader. Also I need to know its resulting public URL in order to spawn it. Everything seems to be perfect with `file-loader`, but:

* I don't want to add `{test: "worker.js", loader: "file?name=[name].[ext]"}` to `webpack.config.js` and use just `const url = require("worker.js")` because then my JS is tightly coupled to build configuration - if I change the worker file name in the souces but forget about the webpack config, everything will broke
* I don't want to set file loader options in JS file either, like this: `const url = require("file?name=[name].[ext]!worker.js")`, because again, my JS will be tightly coupled to the build configuration - I want to change the name based on the `NODE_ENV` and it would be wrong to add that logic to the common JS sources

In other words I want to be able to specify `require("file!worker.js")` in JS file and `{file: {name: "[name].[ext]"}}` somewhere in `webpack.config.js`

I haven't found in the documentation/webpack issues anything similar, but look for example at this [PR](https://github.com/babel/babel-loader/issues/46) to the babel loader. It seems to be that users do want to provide global/default options for loaders sometimes.

My feature request could be easily implemented because loaders have access to the webpack's global config but maybe I'm doing something wrong or don't understand something and would like to hear your opinion on this.

**Comment by aldendaniels** [comment]
:+1: I would also use this feature.

**Comment by Kagami** [comment]
Implemented in #42.
